internal/storage/http.go

```diff
@@ -0,0 +1,78 @@
+package storage
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"time"
+
+	"github.com/nlepage/go-tarfs"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type HTTP struct {
+	client      http.Client
+	requestOpts []func(*http.Request)
+}
+
+type HTTPOption func(*HTTP)
+
+func WithInsecureSkipVerify(v bool) HTTPOption {
+	return func(s *HTTP) {
+		tr := s.client.Transport.(*http.Transport)
+		if tr.TLSClientConfig == nil {
+			tr.TLSClientConfig = &tls.Config{}
+		}
+		tr.TLSClientConfig.InsecureSkipVerify = v
+	}
+}
+
+func WithBearerToken(token string) HTTPOption {
+	return func(s *HTTP) {
+		s.requestOpts = append(s.requestOpts, func(request *http.Request) {
+			request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+		})
+	}
+}
+
+type HTTPRequestOption func(*http.Request)
+
+func NewHTTP(opts ...HTTPOption) *HTTP {
+	s := &HTTP{client: http.Client{
+		Timeout:   time.Minute,
+		Transport: http.DefaultTransport.(*http.Transport).Clone(),
+	}}
+	for _, f := range opts {
+		f(s)
+	}
+	return s
+}
+
+func (s *HTTP) Load(ctx context.Context, owner client.Object) (fs.FS, error) {
+	bundle := owner.(*rukpakv1alpha1.Bundle)
```

Comment:
nit: add a cast check in case the owner isn't a Bundle struct?

Reply:
I'm thinking we might just want to change the storage interface to accept rukpakv1alpha1.Bundle directly.  I don't think we have a use case for storage working with anything other than that. WDYT? Follow-up?

For now, I feel like panicing is better than returning an error because there's nothing an end user can do if this cast fails.

Reply:
I'm +1 to this function taking a `rukpakv1alpha1.Bundle` directly since we're using its properties directly on line 58.